src/select-list.h

```diff
@@ -0,0 +1,34 @@
+/*   string/int GtkListStore for GtkComboBox (display name plus value)
+ *   Copyright (C) 2023 Oskar Berggren
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 2 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program; if not, write to the Free Software
+ *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#ifndef __GTT_SELECT_LIST_H__
+#define __GTT_SELECT_LIST_H__
+
+#include <gtk/gtk.h>
+
+GtkListStore *gtt_select_list_new();
```

Comment:
This has the taste of creating a new type. I wonder if it would be cleaner to go this road entirely. It's just a thought, so feel welcome to leave it as it is.

When thinking about this issue I had the slight inclination to go even further. It seems that the `GtkOptionMenu`  instances are used at least partially to choose enumerators. So for maximum convenience for the trade off of some slight code bloat (here I miss generics in C ;-)) why not create specialized combo box types for the enumerations in question?

Reply:
This is all deprecated again in Gtk 4.10, and the idea in Gtk4+ seems to be to use GtkDropDown and GListModel instead of GtkComboBox and GtkListStore. So for now I aimed for a solution that didn't duplicate too much code and that didn't make things worse (i.e. I wanted to avoid putting just strings in the list and rely on string matching or element order), and then in Gtk 4 there may be a different approach.

I miss generics too. :) Creating dedicated subclasses for the GtkComboBox may be overkill, especially since they are each used in just one place, but I was considering that a specialization to display any enum might be useful. Possibly based on GEnumValue. To be honest this seems like such a common concept that it's strange there is no existing convenience helper for it.

Reply:
So in summary, I think further improvements on this should wait until they can be done based on Gtk4.

Reply:
Sounds fine :+1: